api/src/main/java/org/openmrs/module/ugandaemrreports/reports2019/SetupANCRegister2019.java

```diff
@@ -0,0 +1,224 @@
+import org.openmrs.module.ugandaemrreports.reporting.metadata.Dictionary;
+import org.openmrs.module.ugandaemrreports.reports.UgandaEMRDataExportManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ */
+@Component
+public class SetupANCRegister2019 extends UgandaEMRDataExportManager {
+
+    @Autowired
+    private DataFactory df;
+
+    @Autowired
+    SharedDataDefintion sdd;
+
+    @Autowired
+    BasePatientDataLibrary basePatientDataLibrary;
+
+
+    /**
+     * @return the uuid for the report design for exporting to Excel
+     */
+    @Override
+    public String getExcelDesignUuid() {
+        return "b4720606-74ec-4295-baac-c2f7c774d04f";
+    }
+
+    public String getCSVDesignUuid()
+    {
+        return "79cddc75-d380-439c-a0f2-1911603d1f8e";
+    }
+
+
+    @Override
+    public List<ReportDesign> constructReportDesigns(ReportDefinition reportDefinition) {
+        List<ReportDesign> l = new ArrayList<ReportDesign>();
+        l.add(buildReportDesign(reportDefinition));
+        l.add(buildExcelDesign(reportDefinition));
+        return l;
+    }
+
+    /**
+     * Build the report design for the specified report, this allows a user to override the report design by adding properties and other metadata to the report design
+     *
+     * @param reportDefinition
+     * @return The report design
+     */
+
+    @Override
+    public ReportDesign buildReportDesign(ReportDefinition reportDefinition) {
+        ReportDesign rd = createCSVDesign(getCSVDesignUuid(), reportDefinition);
+        return rd;
+    }
+    public ReportDesign buildExcelDesign(ReportDefinition reportDefinition) {
+        ReportDesign rd = createExcelTemplateDesign(getExcelDesignUuid(), reportDefinition, "ANCRegister2019.xls");
+        Properties props = new Properties();
+        props.put("repeatingSections", "sheet:1,row:10-13,dataset:ANC");
+        props.put("sortWeight", "5000");
+        rd.setProperties(props);
+        return rd;
+    }
+
+    @Override
+    public String getUuid() {
+        return "95b4a987-2a8d-41f2-94b2-9c5bbeef9ca9";
+    }
+
+    @Override
+    public String getName() {
+        return "Intergrated Antenatal Register";
+    }
+
+    @Override
+    public String getDescription() {
+        return "It contains ANC information about the mother";
+    }
+
+    @Override
+    public ReportDefinition constructReportDefinition() {
+        ReportDefinition rd = new ReportDefinition();
+
+        rd.setUuid(getUuid());
+        rd.setName(getName());
+        rd.setDescription(getDescription());
+        rd.addParameters(getParameters());
+        rd.addDataSetDefinition("ANC", Mapped.mapStraightThrough(dataSetDefinition()));
+        return rd;
+    }
+
+    @Override
+    public String getVersion() {
+        return "2.0.9.7";
```

Comment:
@solema only 3 digits not 4, should be 3.0.0

Reply:
done